todo/src/App.js

```diff
@@ -1,28 +1,55 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
 import './App.css';
+import TodoForm from './components/TodoForm';
+import TodoList from './components/TodoList';
+import { connect } from 'react-redux';
+import { addTodo, toggleTodo, clearCompleted } from './actions';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      task: ''
+    }
+  }
+
+  handleChanges = e => {
+    console.log(this.state)
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  addTodo = e => {
+    e.preventDefault();
+    this.props.addTodo(this.state.task);
+    this.setState({ task: '' })
+  }
+
+  toggleTodo = id => {
+    this.props.toggleTodo(id);
+  }
+
+  clearCompleted = () => {
+    this.props.clearCompleted();
+  }
+
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-          <a
-            className="App-link"
-            href="https://reactjs.org"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Learn React
-          </a>
-        </header>
+      <div className="container">
+        <h1>ToDo List</h1>
+        <TodoList todos={this.props.todos} toggleTodo={this.toggleTodo} />
+        <TodoForm handleChanges={this.handleChanges} addTodo={this.addTodo} task={this.state.task} />
```

Comment:
it's fine to have your state and methods held on App and passed down to your TodoForm, but it's not really in the Redux spirit of things which takes away the need to pass around state and props from component to component like this. Since App never does anything else with the input value held on state (it's not used anywhere else in the app) there's no reason for it not to exist solely within the TodoForm component when using Redux. And if another piece of the app _did_ need it for something, then it should exist as part of the state held in the Redux store and be accessed that way. Otherwise if this was a Redux-less React app this would good practice